libp2p/transports/wstransport.nim

```diff
@@ -0,0 +1,199 @@
+## Nim-LibP2P
+## Copyright (c) 2021 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+{.push raises: [Defect].}
+
+import std/[sequtils]
+import chronos, chronicles
+import transport,
+       ../errors,
+       ../wire,
+       ../multicodec,
+       ../multistream,
+       ../connmanager,
+       ../multiaddress,
+       ../stream/connection,
+       ../upgrademngrs/upgrade,
+       websock/websock
+
+logScope:
+  topics = "libp2p wstransport"
+
+export transport
+
+const
+  WsTransportTrackerName* = "libp2p.wstransport"
+
+type
+  WsStream = ref object of Connection
+    session: WSSession
+
+proc init*(T: type WsStream,
+           session: WSSession,
+           dir: Direction,
+           timeout = 10.minutes,
+           observedAddr: MultiAddress = MultiAddress()): T =
```

Comment:
I can't see a need to pass `observedAddr` here. This information is usually only available after we've received a connection, specifically during identify, we _might_ learn about a new address that the remote was able to reach us over. Even if we are able to populate it from the incoming raw http request, it's still something that is internal to the transport.

Reply:
Ah, nevermind - I got confused thinking this is the transport being initialized.